sql/src/main/java/io/crate/analyze/fetch/FetchFieldExtractor.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.fetch;
+
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.AnalyzedRelationVisitor;
+import io.crate.analyze.relations.DocTableRelation;
+import io.crate.analyze.symbol.DefaultTraversalSymbolVisitor;
+import io.crate.analyze.symbol.Field;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.metadata.doc.DocSysColumns;
+
+import java.util.Set;
+
+public class FetchFieldExtractor {
+
+    public static void extract(Iterable<? extends Symbol> symbols, Set<Symbol> fields){
+        for (Symbol symbol : symbols) {
+            FieldVisitor.INSTANCE.process(symbol, fields);
+        }
+    }
+
+    private static class FieldVisitor extends DefaultTraversalSymbolVisitor<Set<Symbol>, Void>{
+
+        private static final FieldVisitor INSTANCE = new FieldVisitor();
+
+
+        @Override
+        public Void visitField(Field field, Set<Symbol> context) {
```

Comment:
is this important that the context is a set or could it also be just a collection so that it works for lists too?